// reactionic/dist/components/ionActionSheet.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _lodash = require('lodash');

var _lodash2 = _interopRequireDefault(_lodash);

var _transitionendProperty = require('transitionend-property');

var _transitionendProperty2 = _interopRequireDefault(_transitionendProperty);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var IonActionSheet = _react2.default.createClass({
  displayName: 'IonActionSheet',

  contextTypes: {
    ionActionSheet: _react2.default.PropTypes.object,
    ionUpdateActionSheet: _react2.default.PropTypes.func
  },
  getInitialState: function getInitialState() {
    return {
      isUp: false,
      callback: function callback() {}
    };
  },
  cancelAction: function cancelAction(e) {
    e && e.stopPropagation();
    this.close(this.context.ionActionSheet.cancel);
  },
  destructiveButtonClicked: function destructiveButtonClicked(e) {
    e && e.stopPropagation();
    this.close(this.context.ionActionSheet.destructiveButtonClicked);
  },
  buttonClicked: function buttonClicked(e, idx) {
    e && e.stopPropagation();
    this.close(this.context.ionActionSheet.buttonClicked.bind(null, idx));
  },
  close: function close(callback) {
    if (this.state.isUp) {
      this.setState({ isUp: false, callback: callback });
    }
  },
  componentDidUpdate: function componentDidUpdate(prevProps, prevState) {
    if (_lodash2.default.isEmpty(prevProps.ionActionSheet) && !_lodash2.default.isEmpty(this.context.ionActionSheet)) {
      // show actionSheet
      this.setState({ isUp: true });
    }
    if (!this.state.isUp && prevState.isUp) {
      var self = this;
      var handler = function handler() {
        self.context.ionUpdateActionSheet({});
        wrapper.removeEventListener(_transitionendProperty2.default, handler);
        if (typeof self.state.callback === 'function') {
          self.state.callback();
        }
      };
      var wrapper = this.refs.wrapper;
      wrapper.addEventListener(_transitionendProperty2.default, handler);
    }
  },
  componentWillUpdate: function componentWillUpdate(nextProps, nextState) {
    if (_lodash2.default.isEmpty(nextProps.ionActionSheet) && !_lodash2.default.isEmpty(this.context.ionActionSheet)) {
      // hide actionSheet
      this.cancelAction(false);
    }
  },
  render: function render() {
    var _this = this;

    var ionActionSheet = this.context.ionActionSheet;

    var willMount = true;
    if (_lodash2.default.isEmpty(ionActionSheet)) willMount = false;

    var titleText = ionActionSheet.titleText;
    var destructiveText = ionActionSheet.destructiveText;
    var cancelText = ionActionSheet.cancelText;
    var buttons = ionActionSheet.buttons;
    var cancel = ionActionSheet.cancel;
    var buttonClicked = ionActionSheet.buttonClicked;
    var destructiveButtonClicked = ionActionSheet.destructiveButtonClicked;
    var ionUpdateActionSheet = this.context.ionUpdateActionSheet;
    var onclickCancel = function onclickCancel(e) {
      _this.cancelAction(e);
    };
    var onclickDelete = function onclickDelete(e) {
      _this.destructiveButtonClicked(e);
    };

    if (titleText) {
      titleText = _react2.default.createElement(
        'div',
        { className: 'action-sheet-title' },
        titleText
      );
    }
    if (buttons) {
      (function () {
        var self = _this;
        buttons = buttons.map(function (button, idx) {
          if (button.text) {
            return _react2.default.createElement(
              'button',
              { className: 'button', key: idx, onClick: function onClick(e) {
                  return self.buttonClicked(e, idx);
                } },
              button.text
            );
          } else {
            return null;
          }
        });
      })();
    }
    if (destructiveText) {
      destructiveText = _react2.default.createElement(
        'div',
        { className: 'action-sheet-group' },
        _react2.default.createElement(
          'button',
          { className: 'button destructive', onClick: onclickDelete },
          destructiveText
        )
      );
    }
    if (cancelText) {
      cancelText = _react2.default.createElement(
        'div',
        { className: 'action-sheet-group' },
        _react2.default.createElement(
          'button',
          { className: 'button', onClick: onclickCancel },
          cancelText
        )
      );
    }

    var backdropClasses = (0, _classnames2.default)({ 'action-sheet-backdrop': willMount, 'active': this.state.isUp });
    var classes = (0, _classnames2.default)({ 'action-sheet-wrapper': true, 'action-sheet-up': this.state.isUp });

    return _react2.default.createElement(
      'div',
      { className: backdropClasses, onClick: onclickCancel },
      _react2.default.createElement(
        'div',
        { className: classes, ref: 'wrapper' },
        _react2.default.createElement(
          'div',
          { className: 'action-sheet' },
          _react2.default.createElement(
            'div',
            { className: 'action-sheet-group' },
            titleText,
            buttons
          ),
          destructiveText,
          cancelText
        )
      )
    );
  }
});

exports.default = IonActionSheet;
