// reactionic/dist/components/ionPopover.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.IonPopoverButton = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _ionButton = require('./ionButton');

var _ionButton2 = _interopRequireDefault(_ionButton);

var _lodash = require('lodash');

var _lodash2 = _interopRequireDefault(_lodash);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var POPOVER_BODY_PADDING = 6;

var IonPopover = _react2.default.createClass({
  displayName: 'IonPopover',

  propTypes: {},
  getDefaultProps: function getDefaultProps() {
    return {};
  },
  contextTypes: {
    ionShowPopover: _react2.default.PropTypes.oneOfType([_react2.default.PropTypes.func, _react2.default.PropTypes.bool])
  },
  backdropClicked: function backdropClicked(e) {
    e.preventDefault();
    if (e.target.className.indexOf("popover-backdrop") >= 0) {
      // if clicked on backdrop outside of the popover, close popover
      this.context.ionShowPopover(false);
    }
  },
  getInitialState: function getInitialState() {
    return {
      popoverBottomClass: false,
      popoverStyle: {},
      arrowStyle: {}
    };
  },
  ignoreClick: function ignoreClick(e) {
    e && e.stopPropagation(); // so it won't close the popover
  },
  componentDidUpdate: function componentDidUpdate(prevProps, prevState) {
    if (this.props.children && _lodash2.default.isEmpty(this.state.popoverStyle)) {
      // some old skool hacks to position the popover after all is mounted
      var htmlElement = document.documentElement;
      var button = document.getElementsByClassName("ionPopoverButton")[0];
      var arrow = this.refs.arrow;
      var popover = this.refs.popover;

      var bodyHeight = htmlElement.clientHeight;
      var bodyWidth = htmlElement.clientWidth;
      var buttonPosition = button.getBoundingClientRect();
      var buttonPositionLeft = buttonPosition.left + document.body.scrollLeft;
      var buttonPositionTop = buttonPosition.top + document.body.scrollTop;
      var buttonWidth = button.offsetWidth;
      var buttonHeight = button.offsetHeight;
      var popoverWidth = popover.offsetWidth;
      var popoverHeight = popover.offsetHeight;

      var popoverCSS = {
        marginLeft: '0',
        opacity: 1,
        left: buttonPositionLeft + buttonWidth / 2 - popoverWidth / 2
      };

      if (popoverCSS.left < POPOVER_BODY_PADDING) {
        popoverCSS.left = POPOVER_BODY_PADDING;
      } else if (popoverCSS.left + popoverWidth + POPOVER_BODY_PADDING > bodyWidth) {
        popoverCSS.left = bodyWidth - popoverWidth - POPOVER_BODY_PADDING;
      }

      if (buttonPositionTop + buttonHeight + popoverHeight > bodyHeight) {
        popoverCSS.top = buttonPositionTop - popoverHeight;
        this.setState({ popoverBottomClass: true });
      } else {
        popoverCSS.top = buttonPositionTop + buttonHeight;
        this.setState({ popoverBottomClass: false });
      }

      this.setState({ arrowStyle: { left: buttonPositionLeft + buttonWidth / 2 - arrow.offsetWidth / 2 - popoverCSS.left } });
      this.setState({ popoverStyle: popoverCSS });
    }
  },
  render: function render() {
    var popOverClasses = (0, _classnames2.default)({ 'popover': true,
      'popover-bottom': this.state.popoverBottomClass });
    var backdropClasses = (0, _classnames2.default)({ 'popover-backdrop': true,
      'active': this.props.children });
    var content = null;
    if (this.props.children) {
      content = _react2.default.createElement(
        'div',
        { className: backdropClasses, onClick: this.backdropClicked },
        _react2.default.createElement(
          'div',
          { className: 'popover-wrapper', onClick: this.ignoreClick },
          _react2.default.createElement(
            'div',
            { className: popOverClasses, style: this.state.popoverStyle, ref: 'popover' },
            _react2.default.createElement('div', { className: 'popover-arrow', style: this.state.arrowStyle, ref: 'arrow' }),
            this.props.children
          )
        )
      );
    }
    return _react2.default.createElement(
      'div',
      null,
      content
    );
  }
});

var IonPopoverButton = _react2.default.createClass({
  displayName: 'IonPopoverButton',
  render: function render() {
    return _react2.default.createElement(_ionButton2.default, _extends({ customClasses: 'ionPopoverButton' }, this.props));
  }
});

exports.default = IonPopover;
exports.IonPopoverButton = IonPopoverButton;
