// reactionic/dist/components/ionSubFooterBar.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _reactRouter = require('react-router');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var IonSubFooterBar = _react2.default.createClass({
  displayName: 'IonSubFooterBar',

  propTypes: {
    customClasses: _react2.default.PropTypes.string
  },
  getDefaultProps: function getDefaultProps() {
    return {
      customClasses: ''
    };
  },
  contextTypes: {
    ionUpdateHasX: _react2.default.PropTypes.func
  },
  componentWillMount: function componentWillMount() {
    this.context.ionUpdateHasX('ionHasSubfooter', true);
  },
  mixins: [_reactRouter.Lifecycle],
  routerWillLeave: function routerWillLeave(nextLocation) {
    this.context.ionUpdateHasX('ionHasSubfooter', false);
  },
  render: function render() {
    var classes = (0, _classnames2.default)({ 'bar': true, 'bar-subfooter': true }, this.props.customClasses || 'bar-stable' // default class
    );
    return _react2.default.createElement(
      'div',
      { className: classes },
      this.props.children
    );
  }
});

exports.default = IonSubFooterBar;
